src/displayapp/screens/settings/SettingSetTime.cpp

```diff
@@ -0,0 +1,160 @@
+  lv_obj_set_event_cb(btnHoursPlus, event_handler);
+
+  btnHoursMinus = lv_btn_create(lv_scr_act(), NULL);
+  btnHoursMinus->user_data = this;
+  lv_obj_set_size(btnHoursMinus, 50, 40);
+  lv_obj_align(btnHoursMinus, lv_scr_act(), LV_ALIGN_CENTER, -72, 40);
+  lv_obj_set_style_local_value_str(btnHoursMinus, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "-");
+  lv_obj_set_event_cb(btnHoursMinus, event_handler);
+
+  btnMinutesPlus = lv_btn_create(lv_scr_act(), NULL);
+  btnMinutesPlus->user_data = this;
+  lv_obj_set_size(btnMinutesPlus, 50, 40);
+  lv_obj_align(btnMinutesPlus, lv_scr_act(), LV_ALIGN_CENTER, 0, -50);
+  lv_obj_set_style_local_value_str(btnMinutesPlus, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "+");
+  lv_obj_set_event_cb(btnMinutesPlus, event_handler);
+
+  btnMinutesMinus = lv_btn_create(lv_scr_act(), NULL);
+  btnMinutesMinus->user_data = this;
+  lv_obj_set_size(btnMinutesMinus, 50, 40);
+  lv_obj_align(btnMinutesMinus, lv_scr_act(), LV_ALIGN_CENTER, 0, 40);
+  lv_obj_set_style_local_value_str(btnMinutesMinus, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "-");
+  lv_obj_set_event_cb(btnMinutesMinus, event_handler);
+
+  btnSetTime = lv_btn_create(lv_scr_act(), NULL);
+  btnSetTime->user_data = this;
+  lv_obj_set_size(btnSetTime, 120, 48);
+  lv_obj_align(btnSetTime, lv_scr_act(), LV_ALIGN_IN_BOTTOM_MID, 0, 0);
+  lv_obj_set_style_local_value_str(btnSetTime, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "Set");
+  lv_obj_set_event_cb(btnSetTime, event_handler);
+}
+
+SettingSetTime::~SettingSetTime() {
+  lv_obj_clean(lv_scr_act());
+}
+
+bool SettingSetTime::Refresh() {
+  return running;
+}
+
+
+void SettingSetTime::HandleButtonPress(lv_obj_t *object, lv_event_t event) {
+
+  if (event != LV_EVENT_CLICKED)
+    return;
+
+  if (object == btnHoursPlus) {
+    hoursValue++;
+    if (hoursValue > 23)
+      hoursValue = 0;
+    lv_label_set_text_fmt(lblHours, "%02d", hoursValue);
+    lv_obj_align(lblHours, lv_scr_act(), LV_ALIGN_CENTER, -72, -6);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+  }
+  else if (object == btnHoursMinus) {
+    hoursValue--;
+    if (hoursValue < 0)
+      hoursValue = 23;
+    lv_label_set_text_fmt(lblHours, "%02d", hoursValue);
+    lv_obj_align(lblHours, lv_scr_act(), LV_ALIGN_CENTER, -72, -6);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+  }
+  else if (object == btnMinutesPlus) {
+    minutesValue++;
+    if (minutesValue > 59)
+      minutesValue = 0;
+    lv_label_set_text_fmt(lblMinutes, "%02d", minutesValue);
+    lv_obj_align(lblMinutes, lv_scr_act(), LV_ALIGN_CENTER, 0, -6);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+  }
+  else if (object == btnMinutesMinus) {
+    minutesValue--;
+    if (minutesValue < 0)
+      minutesValue = 59;
+    lv_label_set_text_fmt(lblMinutes, "%02d", minutesValue);
+    lv_obj_align(lblMinutes, lv_scr_act(), LV_ALIGN_CENTER, 0, -6);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+  }
```

Comment:
It should be possible to use `lv_obj_realign(obj)` or `lv_obj_set_auto_realign(obj, true)` here and in the other file, so the position doesn't need to be updated. Also in this file the positions aren't set by `#define`

Reply:
`lv_obj_set_auto_realign` worked, so I changed the code in both files accordingly.
The positions in `SettingSetTime` are now determined by `#define`s.

Btw, thanks for your feedback. I usually code using Qt and this is my "first contact" with LVGL...